From: Vojtech Trefny <vtrefny@redhat.com>
Date: Mon, 9 Mar 2026 12:25:45 +0100
Subject: tests: Lower the expected amount of free space in ext filesystems

With e2fsprogs 1.47.4-1 there are some changes to calculating the
size of the orphan inode which makes the tests fail. We don't
really need to check the exact value, just make sure we return
something that makes sense.

Fixes: #1179
---
 tests/fs_tests/ext_test.py | 14 ++++----------
 1 file changed, 4 insertions(+), 10 deletions(-)

diff --git a/tests/fs_tests/ext_test.py b/tests/fs_tests/ext_test.py
index 790f692..b0195ec 100644
--- a/tests/fs_tests/ext_test.py
+++ b/tests/fs_tests/ext_test.py
@@ -243,8 +243,8 @@ class ExtGetInfo(ExtTestCase):
         self.assertTrue(fi)
         self.assertEqual(fi.block_size, 1024)
         self.assertEqual(fi.block_count, self.loop_size / 1024)
-        # at least 90 % should be available, so it should be reported
-        self.assertGreater(fi.free_blocks, 0.90 * self.loop_size / 1024)
+        # at least 50 % should be available, so it should be reported
+        self.assertGreater(fi.free_blocks, 0.50 * self.loop_size / 1024)
         self.assertEqual(fi.label, "")
         # should be an non-empty string
         self.assertTrue(fi.uuid)
@@ -254,8 +254,8 @@ class ExtGetInfo(ExtTestCase):
             fi = BlockDev.fs_ext4_get_info(self.loop_devs[0])
             self.assertEqual(fi.block_size, 1024)
             self.assertEqual(fi.block_count, self.loop_size / 1024)
-            # at least 90 % should be available, so it should be reported
-            self.assertGreater(fi.free_blocks, 0.90 * self.loop_size / 1024)
+            # at least 50 % should be available, so it should be reported
+            self.assertGreater(fi.free_blocks, 0.50 * self.loop_size / 1024)
             self.assertEqual(fi.label, "")
             # should be an non-empty string
             self.assertTrue(fi.uuid)
@@ -344,8 +344,6 @@ class ExtResize(ExtTestCase):
         self.assertTrue(fi)
         self.assertEqual(fi.block_size, 1024)
         self.assertEqual(fi.block_count, self.loop_size / 1024)
-        # at least 90 % should be available, so it should be reported
-        self.assertGreater(fi.free_blocks, 0.90 * self.loop_size / 1024)
 
         succ = resize_function(self.loop_devs[0], 50 * 1024**2, None)
         self.assertTrue(succ)
@@ -361,8 +359,6 @@ class ExtResize(ExtTestCase):
         self.assertTrue(fi)
         self.assertEqual(fi.block_size, 1024)
         self.assertEqual(fi.block_count, self.loop_size / 1024)
-        # at least 90 % should be available, so it should be reported
-        self.assertGreater(fi.free_blocks, 0.90 * self.loop_size / 1024)
 
         # resize again
         succ = resize_function(self.loop_devs[0], 50 * 1024**2, None)
@@ -379,8 +375,6 @@ class ExtResize(ExtTestCase):
         self.assertTrue(fi)
         self.assertEqual(fi.block_size, 1024)
         self.assertEqual(fi.block_count, self.loop_size / 1024)
-        # at least 90 % should be available, so it should be reported
-        self.assertGreater(fi.free_blocks, 0.90 * self.loop_size / 1024)
 
         # get min size and resize to it
         size = minsize_function(self.loop_devs[0])
